docs/ytdl.rst

```diff
@@ -0,0 +1,404 @@
+            'subtitles': {},
+            'automatic_captions': {},
+            'duration': 141,
+            'age_limit': 0,
+            'annotations': None,
+            'chapters': None,
+            'webpage_url': 'https://www.youtube.com/watch?v=-tvs8a_hOQo',
+            'view_count': 1866047,
+            'like_count': 37496,
+            'dislike_count': 422,
+            'average_rating': 4.95548295975,
+            'formats': [...],
+            'is_live': None,
+            'start_time': None,
+            'end_time': None,
+            'series': None,
+            'season_number': None,
+            'episode_number': None,
+            'extractor': 'youtube',
+            'webpage_url_basename': 'watch',
+            'extractor_key': 'Youtube'
+        }
+
+
+A typical youtube format entry looks like this.  There are usually a large number of these representing the various formats you can access a YouTube video in.
+
+.. details:: Click to expand
+
+    .. code-block:: python3
+
+        {
+            "format_id": "18",
+            "url": "https://....googlevideo.com/videoplayback?...",
+            "player_url": None,
+            "ext": "mp4",
+            "width": 640,
+            "height": 360,
+            "acodec": "mp4a.40.2",
+            "abr": 96,
+            "vcodec": "avc1.42001E",
+            "resolution": "640x360",
+            "format_note": "medium",
+            "format": "18 - 640x360 (medium)",
+            "protocol": "https",
+            "http_headers": {<standard http headers>}
+        }
+
+
+Additional Information
+-----------------------
+
+Now that you understand the basics of how youtube-dl works, lets get on to some more involved topics.
+
+Handling Playlists
+~~~~~~~~~~~~~~~~~~~
+
+Dealing with single media requests is fairly straightforward, but playlists require more attention.  For example: downloading a large youtube playlist may take a very long time and will block for the duration.  Trying to download an entire playlist in one go is extremely slow and the ``extract_info`` function will not return until it's finished.  Furthermore, just the processing of every playlist item can be very slow for large playlists.  On average, processing a youtube video can take anywhere from about 0.5 to 1.5 seconds per item.
+
+Taking this long is a bit of a problem, even when wrapping the function as async.  Fortunately, this can be avoided by handling playlists as a series of single media items.
+
+Since downloading an entire playlist is slow, lets not do that.  Furthermore, since processing every item is also slow, lets not do that either.  Calling ``extract_info`` with ``process=False``  will return a sparse info dict (quite quickly too).  There's too much you can do with this, but for playlists there's one important thing.  Most extractors will return playlist items in the info dict as a list of small dicts with a ``url`` key.  Just iterate over each one of those instead of all at once.
```

Comment:
I assume `"There's too much you can do with this,"` should be `"There's not too much you can do with this,"`